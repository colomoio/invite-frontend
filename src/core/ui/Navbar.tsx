import {
  Flex,
  useDisclosure,
  Stack,
  Link,
  Heading,
  Image,
} from "@chakra-ui/react";
import NextLink from "next/link";
import { HamburgerIcon } from "@chakra-ui/icons";

export function Navbar() {
  const { isOpen, onOpen, onClose } = useDisclosure();

  return (
    <nav>
      <Flex
        p=".5rem 1rem .5rem .5rem"
        direction={["column", "row"]}
        justify="space-between"
        bg="purple.600"
        borderRadius="0 0 1.5rem 1.5rem"
        transition="3ms"
      >
        <Flex alignItems="center" wrap="wrap" justify="space-between">
          <Image
            borderRadius="full"
            boxSize="2.5rem"
            src="https://bit.ly/dan-abramov"
            alt="app-icon"
          />

          <Heading ml={[isOpen ? "0" : "0", "2rem"]} textColor="white">
            Event App
          </Heading>

          <HamburgerIcon
            onClick={isOpen ? onClose : onOpen}
            display={["inline", "none"]}
            textColor="white"
            boxSize="1.25rem"
            ml="1.25rem"
          />
        </Flex>

        <Flex
          justifyContent={[isOpen ? "center" : "none", "center"]}
          display={[isOpen ? "flex" : "none", "flex"]}
          bg="purple.600"
        >
          <Stack align="center" direction={["column", "row"]}>
            <Link
              marginTop={[isOpen ? ".5rem" : "none", "0"]}
              fontSize="lg"
              fontWeight="medium"
              textColor="white"
              as={NextLink}
              href="/"
              _hover={{
                transform: "scale(1.1)",
                textDecor: "none",
              }}
            >
              Home
            </Link>
            <Link
              fontSize="lg"
              fontWeight="medium"
              textColor="white"
              as={NextLink}
              href="/create"
              _hover={{
                transform: "scale(1.1)",
                textDecor: "none",
              }}
            >
              Create
            </Link>
          </Stack>
        </Flex>
      </Flex>
    </nav>
  );
}
