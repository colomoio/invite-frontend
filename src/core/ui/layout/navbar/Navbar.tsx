import { Flex, Box, useDisclosure, Stack, Link } from "@chakra-ui/react";
import { HamburgerIcon } from "@chakra-ui/icons";

export function Navbar() {
  const { isOpen, onOpen, onClose } = useDisclosure();

  return (
    <div className="App">
      <Flex
        py={2}
        px={4}
        direction={["column", "row"]}
        justify="space-between"
        bg="blue.100"
      >
        <Flex alignItems="center" wrap="wrap">
          <Flex flexGrow={1} justify="center" /*border='1px'*/>
            {/* <Heading ml={[4, 0]}>Zensurance</Heading> */}
            <Box ml={[4, 0]}>Box 1</Box>
          </Flex>
          <HamburgerIcon
            onClick={isOpen ? onClose : onOpen}
            display={["inline", "none"]}
            // border='1px'
          />
        </Flex>
        <Flex display={[isOpen ? "flex" : "none", "flex"]} bg="blue.100">
          <Stack align="center" direction={["column", "row"]}>
            <Link href="/">Home</Link>
            <Link href="/create">create</Link>
            <Link href="/about-us">about us</Link>
          </Stack>
        </Flex>
      </Flex>
    </div>
  );
}
