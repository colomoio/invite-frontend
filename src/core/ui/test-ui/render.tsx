import {
  Fragment,
  FunctionComponent,
  type ReactElement,
  type ReactNode,
} from "react";
import {
  render as rtlRender,
  renderHook as rtlRenderHook,
  type RenderOptions,
  type RenderHookOptions,
} from "@testing-library/react";

import {
  AuthContext,
  type AuthContextType,
} from "../../../users/ui/AuthProvider";

/**
 *
 */
function renderWithAuth(children: ReactNode, initialState: AuthContextType) {
  return (
    <AuthContext.Provider value={initialState}>{children}</AuthContext.Provider>
  );
}

function Render() {
  let rendered: ReactNode = null;

  return function Render(Component: ReactNode) {
    rendered = Component;
    return {
      withAuth(initialState: AuthContextType) {
        rendered = renderWithAuth(rendered, initialState);

        return this;
      },
      now(opts?: RenderOptions) {
        return rtlRender(rendered as ReactElement, opts);
      },
    };
  };
}

type ElementProps = {
  children?: ReactNode;
};

function getWrapper(
  Wrapper: FunctionComponent<ElementProps>,
  render: (children: ReactNode) => JSX.Element
) {
  return function WrapComponent({ children }: ElementProps) {
    return <Wrapper>{render(children)}</Wrapper>;
  };
}

/**
 * @example
 * render(<Component />).withAuth({ user: null }).now()
 */
export const render = Render();

export function renderHook<Props extends any, Result extends unknown>(
  hookFn: (props: Props) => Result
) {
  let Wrapper = ({ children }: ElementProps) => <Fragment>{children}</Fragment>;

  return {
    withAuth(initialState: AuthContextType) {
      Wrapper = getWrapper(Wrapper, (children) =>
        renderWithAuth(children, initialState)
      );

      return this;
    },
    now(opts?: RenderHookOptions<Props>) {
      return rtlRenderHook(hookFn, {
        wrapper: ({ children }) => <Wrapper>{children}</Wrapper>,
        ...opts,
      });
    },
  };
}
