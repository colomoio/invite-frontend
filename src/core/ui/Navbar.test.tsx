import { screen } from "@testing-library/react";
import { Navbar } from "./Navbar";
import { render } from "./test-ui/render";

describe("Navbar", () => {
  it("renders", () => {
    const { asFragment } = render(<Navbar />).now();
    expect(asFragment()).toMatchSnapshot();
  });
  it("displays routes", () => {
    render(<Navbar />).now();
    expect(screen.getByText("Home")).toBeVisible();
    expect(screen.getByText("Create")).toBeVisible();
  });
});
