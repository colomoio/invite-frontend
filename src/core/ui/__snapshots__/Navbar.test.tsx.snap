// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders 1`] = `
<DocumentFragment>
  <nav>
    <div
      class="css-tn3qd3"
    >
      <div
        class="css-1dtnjt5"
      >
        <div
          class="css-2a4gch"
        >
          <h2
            class="chakra-heading css-132pka0"
          >
            Event App
          </h2>
        </div>
        <svg
          class="chakra-icon css-fv7h53"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="css-1bpfvr5"
      >
        <div
          class="chakra-stack css-gwxmxi"
        >
          <a
            class="chakra-link css-h5k7go"
            href="/"
          >
            Home
          </a>
          <a
            class="chakra-link css-1sdkjyt"
            href="/create"
          >
            Create
          </a>
        </div>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
